pkg/steps/source.go

```diff
@@ -482,6 +500,17 @@ func handleBuilds(ctx context.Context, buildClient BuildClient, podClient kubern
 		o = opts[0]
 	}
 
+	// Create closure that checks if this build is for a multi-arch repo
+	needsMultiArchWorkaround := o.NeedsMultiArchWorkaround
+	if needsMultiArchWorkaround == nil {
+		needsMultiArchWorkaround = func() bool {
```

Comment:
If this closure doesn't take any dynamic data as input, I don't understand why calculating the build's labels later would help? What's the reasoning behind making this a closure?